ndonnx/_funcs.py

```diff
@@ -23,6 +20,10 @@
     promote,
 )
 
+
+class UnsupportedOperationError(TypeError): ...
```

Comment:
Just checking: `TypeError` is what we want as a parent? Not `ValueError` or something else? This ends up being in the public API.

Reply:
I suppose `ValueError` would be more accurate as the dtype is a value? I can change it, thanks.